codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/types/StructureGenerator.java

```diff
@@ -90,13 +96,26 @@ public void render() {
         writeRbs(writer -> {
             String shapeName = symbolProvider.toSymbol(shape).getName();
             writer
-                    .openBlock("class $L < ::Struct[untyped]", shapeName)
+                    .openBlock("class $L", shapeName)
                     .write("include $L", Hearth.STRUCTURE)
                     .call(() -> {
                         shape.members().forEach(memberShape -> {
-                            String name = symbolProvider.toMemberName(memberShape);
-                            Symbol target = symbolProvider.toSymbol(model.expectShape(memberShape.getTarget()));
-                            writer.write("attr_accessor $L (): $L", name, target.getProperty("rbsType").get());
+                            String memberName = symbolProvider.toMemberName(memberShape);
+                            Shape target = model.expectShape(memberShape.getTarget());
+                            Symbol symbol = symbolProvider.toSymbol(target);
+                            String rbsType;
+                            // TODO: not sure if any of this belongs here
+                            if (target.hasTrait(StreamingTrait.class)) {
+                                // TODO: determine if input or output and set accordingly
+                                // String is a hack because of protocol tests comparing blob? FIXME
```

Comment:
Where do we compare the blob in the protocol tests?  

I think on [output](https://github.com/smithy-lang/smithy-ruby/blob/main/codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/HttpProtocolTestGenerator.java#L157) we handle streaming correctly by reading.

But I guess on input when we create our [param inputs](https://github.com/smithy-lang/smithy-ruby/blob/main/codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/util/ParamsToHash.java#L83) we don't check for streaming... maybe we need to there?

Reply:
I can't find a better way around this.